plugin/src/py/android_screenshot_tests/pull_screenshots.py

```diff
@@ -556,15 +556,34 @@ def main(argv):
     multiple_devices = opts.get('--multiple-devices')
     device_calculator = DeviceNameCalculator() if multiple_devices else NoOpDeviceNameCalculator()
 
-    return pull_screenshots(process,
-                            perform_pull=should_perform_pull,
-                            temp_dir=opts.get('--temp-dir'),
-                            filter_name_regex=opts.get('--filter-name-regex'),
-                            opt_generate_png=opts.get('--generate-png'),
-                            record=opts.get('--record'),
-                            verify=opts.get('--verify'),
-                            adb_puller=SimplePuller(puller_args),
-                            device_name_calculator=device_calculator)
+    if "-s" in opts:
+        passed_serials = [opts['-s']]
+    elif "ANDROID_SERIAL" in os.environ:
+        passed_serials = os.environ.get('ANDROID_SERIAL').split(",")
+    else:
+        passed_serials = None
+
+    if passed_serials is not None:
+        for serial in passed_serials:
+            pull_screenshots(process,
+                             perform_pull=should_perform_pull,
+                             temp_dir=opts.get('--temp-dir'),
+                             filter_name_regex=opts.get('--filter-name-regex'),
+                             opt_generate_png=opts.get('--generate-png'),
+                             record=opts.get('--record'),
+                             verify=opts.get('--verify'),
+                             adb_puller=SimplePuller(puller_args + ["-s", serial]),
+                             device_name_calculator=device_calculator)
+    else:
+        pull_screenshots(process,
```

Comment:
I'm not familiar with Python and I couldn't find a way not to have two `pull_screenshots` calls

Reply:
You can flip the problem a bit: create a list of puller args, loop over the puller args

```
puller_arg_list = [puller_args]
...
for serial in passed_serials:
    puller_arg_list = puller_arg_list + (puller_args + ["-s", serial])
...
for puller_args in puller_arg_list:
    pull_screenshots(...
```